adcm_aio_client/core/objects/_imports.py

```diff
@@ -1 +1,33 @@
-class ClusterImports: ...
+from typing import TYPE_CHECKING, Collection, Self, Union
+
+from adcm_aio_client.core.types import Endpoint, Requester
+
+if TYPE_CHECKING:
+    from adcm_aio_client.core.objects.cm import Cluster, Service
+
+
+class Imports:
+    def __init__(self: Self, requester: Requester, path: Endpoint, imports_data: dict) -> None:
+        self._requester = requester
+        self._path = path
+        self._data = imports_data
+
+    async def _get_source_binds(self: Self) -> list[Union["Cluster", "Service"]]:
+        from adcm_aio_client.core.objects.cm import Cluster, Service
+
+        source_types = {"cluster": Cluster, "service": Service}
+        source_binds = [(b["id"], b["type"]) for b in self._data["binds"]["source"]]
+        return [source_types[s[1]](id=s[0], requester=self._requester) for s in source_binds]
```

Comment:
you don't need full objects to detect what the request should contain

Just simplify input `sources` into tuple of type and id AND simplify binds here as well (as you did in line 19)

Then you can use `set` operations to detect `add` and `remove` operation results (code like `required_imports = [i for i in current_imports if i not in sources]` will be more efficient and straightforward)